cpp/include/IceUtil/PushDisableWarnings.h

```diff
@@ -15,11 +15,13 @@
 #    pragma warning(disable : 4251) // class ... needs to have dll-interface to be used by clients of class ...
 #    pragma warning(disable : 4512) //  ... assignment operator could not be generated
 #    pragma warning(disable : 4275) // non dll-interface class ... used as base for dll-interface class ...
+#    pragma warning(disable : 4996) // allow referencing deprecated Slice definitions
```

Comment:
Do we need the equivalent undo  in PopDisableWarnings? We do have any deprecated Slice definitions?

Reply:
`PushDisableWarnings,h` pushes these disablings into a separate 'frame',
so all `PopDisableWarnings` has to do is pop that frame off the stack:
```
#if defined(_MSC_VER)
#    pragma warning(pop)
#elif defined(__clang__)
#    pragma clang diagnostic pop
#elif defined(__GNUC__)
#    pragma GCC diagnostic pop
#endif
```
That is literally the entire file (`PopDisableWarnings.h`)

Reply:
The comment is not correct.

It should include the warning message we're suppressing, and I am pretty sure it doesn't include "Slice definitions".
I assume we're referencing _generated_ C++ types and possibly data members / functions.

Reply:
My comment was aimed at explaining _why_ we needed it, but I changed it to just be the warning message, like we do for other ones.